### ISSUE TITLE
The "stub_domains" settings applied in the wrong cluster when two clusters created.

### ISSUE THREAD

**Issue by movergan** [terraform-google-modules/terraform-google-kubernetes-engine#1835]
### TL;DR

When creating several clusters by this module, 'kube-dns' config map i applied to the default cluster that is specified in Kubernetes provider, 

### Expected behavior

All resources within the module should be created IN the cluster the module creates 

### Observed behavior

kube-dns config map is patched in the wrong cluster. i.e. "gke-test-1" gets values from:

  stub_domains = {
    "example-2.com" = [
      "10.10.10.3",
      "10.10.10.4"
    ]
  }

### Terraform Configuration

```hcl
main.tf:

# google_client_config and kubernetes provider must be explicitly specified like the following.
data "google_client_config" "default" {}

provider "kubernetes" {
  host                   = "https://${module.gke.endpoint}"
  token                  = data.google_client_config.default.access_token
  cluster_ca_certificate = base64decode(module.gke.ca_certificate)
}

provider "kubernetes" {
  alias                  = "kubernetes-2"
  host                   = "https://${module.gke-2.endpoint}"
  token                  = data.google_client_config.default.access_token
  cluster_ca_certificate = base64decode(module.gke-2.ca_certificate)
}

module "gke" {
  source                     = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster"
  project_id                 = "<PROJECT ID>"
  name                       = "gke-test-1"
  region                     = "us-central1"
  zones                      = ["us-central1-a", "us-central1-b", "us-central1-f"]
  
  stub_domains = {
    "example.com" = [
      "10.10.10.1",
      "10.10.10.2"
    ]
  }  
...
}

module "gke-2" {
  source                     = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster"
  project_id                 = "<PROJECT ID>"
  name                       = "gke-test-2"
  region                     = "us-central1"
  zones                      = ["us-central1-a", "us-central1-b", "us-central1-f"]

  stub_domains = {
    "example-2.com" = [
      "10.10.10.3",
      "10.10.10.4"
    ]
  }
...
}
```
```


### Terraform Version

```sh
tf version
Terraform v1.3.1
on linux_amd64
+ provider registry.terraform.io/hashicorp/external v2.3.2
+ provider registry.terraform.io/hashicorp/google v5.10.0
+ provider registry.terraform.io/hashicorp/google-beta v5.10.0
+ provider registry.terraform.io/hashicorp/helm v2.12.1
+ provider registry.terraform.io/hashicorp/kubernetes v2.24.0
+ provider registry.terraform.io/hashicorp/local v2.4.1
+ provider registry.terraform.io/hashicorp/null v3.2.2
+ provider registry.terraform.io/hashicorp/random v3.6.0
+ provider registry.terraform.io/hashicorp/template v2.2.0
```

### Additional information

There is a workaround, to add provider explicitly:

```
module "gke-2" {
  source                     = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster"
  project_id                 = "<PROJECT ID>"
  
    providers = {
    kubernetes = kubernetes.kubernetes-2
  }
  
  name                       = "gke-test-2"
```

I am not sure if it is supposed to be this way but it is extremely confusing and quite dangerous.
